README.md

```diff
@@ -1,133 +1,114 @@
-- the keyname holding the new transaction details tensor
-- a time range, represented by two timestamps.
-
-The function then executes a range query over the sorted set ([`ZRANGEBYSCORE`](https://redis.io/commands/zrangebyscore)) and retrieves a list of hash names (recall that each hash has a corresponding tensor as described in Flow 1).
-
-![Gears<->Redis data gathering](./flow3.png "Gears<->Redis data gathering")
-
-From this list it extracts a list of tensors from the keyspace and sends it to a [Torch script](https://github.com/RedisAI/FraudDetectionDemo/blob/master/app/script.torch). This torch script creates a
-new tensor with the shape `(1, 256)`, by concatenating the tensors and either pad the remaining space or trimming it.
-The new tensor is the reference data for the models which expect a reference data with shape of `(1, 256)` and transaction details tensor with the shape of `(1, 30)`. Once the inferencing of the models are done, the function uses `numpy` to aggregate the results and save them to a tensor
-with shape `(1, 2)` that contains the probability for the transaction to be a fraud.
-
-![Gears<->AI execution](./flow4.png "Gears<->AI execution")
+This demo showcases the advantage of **data locality** during transaction scoring, by using [RedisAI](https://oss.redislabs.com/redisai/) module.
+It simulates a fraud-detection app which is based on ML model prediction in real-time. When a new transaction is executed, it triggers a flow whose output is the probability that this transaction is fraudulent. The prediction is based on both the given transaction and some reference data stored in Redis, while *the entire flow* is done within Redis as well.   
 
 ## Running the Demo
-To run the demo:
+To run the demo app and load the data, run the following:
 ```
-$ git clone https://github.com/RedisAI/FraudDetectionDemo.git
-$ cd FraudDetectionDemo
 # If you don't have it already, install https://git-lfs.github.com/ (On OSX: brew install git-lfs)
 $ git lfs install && git lfs fetch && git lfs checkout
+
+# Clone the repository
+$ git clone https://github.com/RedisAI/FraudDetectionDemo.git
+$ cd FraudDetectionDemo
+
+# Launch the demo with docker-compose (may require sudo in linux)
 $ docker-compose up
 ```
 If something went wrong, e.g. you skipped installing git-lfs, you need to force docker-compose to rebuild the containers
 ```
 $ docker-compose up --force-recreate --build
 ```
 
-The `app` container will [load](https://github.com/RedisAI/FraudDetectionDemo/blob/master/app/app.py)
-- The [two tensorflow models](https://github.com/RedisAI/FraudDetectionDemo/tree/master/app/models)
-- The [Torch script](https://github.com/RedisAI/FraudDetectionDemo/blob/master/app/script.torch)
-- The RedisGears functions
+## Architecture
+### RedisAI
+Redis instance is launched with RedisAI module loaded (by running `redislabs/redisai:latest` docker). RedisAI allows you to store and execute AI/ML pre-trained models in Redis.
```

Comment:
....execute AI/ML models in Redis. The models can be written in Pytorch, Tensorflow or any other ML framework (XGBoot, scikit-learn) allowing models to be exported to ONNX